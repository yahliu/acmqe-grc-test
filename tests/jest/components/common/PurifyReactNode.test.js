/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'
import purifyReactNode from '../../../../src-web/components/common/PurifyReactNode'

describe('test purifyReactNode', () => {
  it('renders null node', () => {
    expect(purifyReactNode(null)).toMatchSnapshot()
  })
  it('renders empty node', () => {
    expect(purifyReactNode('')).toMatchSnapshot()
  })
  it('renders string', () => {
    expect(purifyReactNode('Red Hat')).toMatchSnapshot()
  })
  it('renders number', () => {
    expect(purifyReactNode(2020)).toMatchSnapshot()
  })
  it('renders empty object', () => {
    expect(purifyReactNode({})).toMatchSnapshot()
  })
  it('renders empty array', () => {
    expect(purifyReactNode([])).toMatchSnapshot()
  })
  it('renders object1', () => {
    const object1 = {
      props: {
        name: 'object1'
      }
    }
    expect(purifyReactNode(object1)).toMatchSnapshot()
  })
  it('renders object2', () => {
    const object2 = {
      props: {
        name: 'object2',
        children: 'object2child1'
      }
    }
    expect(purifyReactNode(object2)).toMatchSnapshot()
  })
  it('renders object3', () => {
    const object3 = {
      props: {
        name: 'object3',
        children: {
          props: {
            children: [
              'object3->child1',
              {
                props: {
                  name: 'object3->child1',
                  children: 'object3->child1->grandchild1'
                }
              }
            ]
          }
        }
      }
    }
    expect(purifyReactNode(object3)).toMatchSnapshot()
  })
  it('renders object4', () => {
    const object4 = {
      props: {
        name: 'object4',
        children: {
          props: {
            children: [
              'object4->child1',
              {
                props: {
                  'text': 'NonCompliant; violation - namespaces `prod` does not exist as specified',
                  'maxCharacters': 300,
                }
              }
            ]
          }
        }
      }
    }
    expect(purifyReactNode(object4)).toMatchSnapshot()
  })
})
