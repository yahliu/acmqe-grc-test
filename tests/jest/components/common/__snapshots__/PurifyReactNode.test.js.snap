// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test purifyReactNode renders empty array 1`] = `""`;

exports[`test purifyReactNode renders empty node 1`] = `""`;

exports[`test purifyReactNode renders empty object 1`] = `""`;

exports[`test purifyReactNode renders null node 1`] = `""`;

exports[`test purifyReactNode renders number 1`] = `"2020"`;

exports[`test purifyReactNode renders object1 1`] = `""`;

exports[`test purifyReactNode renders object2 1`] = `"object2child1"`;

exports[`test purifyReactNode renders object3 1`] = `"object3->child1object3->child1->grandchild1"`;

exports[`test purifyReactNode renders object4 1`] = `"object4->child1NonCompliant; violation - namespaces \`prod\` does not exist as specified"`;

exports[`test purifyReactNode renders string 1`] = `"Red Hat"`;
